packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,204 @@
+    /// @param _livenessResponsePeriod The period in seconds for a liveness response
+    /// @param _fallbackOwner The address that will become owner if challenge succeeds
+    function enableModule(uint256 _livenessResponsePeriod, address _fallbackOwner) external {
+        if (_livenessResponsePeriod == 0 || _fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
+
+        // Set the caller as a safe and store its configuration
+        ModuleConfig storage config = safeConfigs[msg.sender];
+
+        // Store the parameters related to this safe
+        config.livenessResponsePeriod = _livenessResponsePeriod;
+        config.fallbackOwner = _fallbackOwner;
+
+        // Clear any existing challenge when enabling/re-enabling
+        delete challengeStartTime[msg.sender];
+
+        emit ModuleEnabled(msg.sender, _livenessResponsePeriod, _fallbackOwner);
+    }
+
+    /// @notice Disables the module by an enabled safe
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST erase the existing liveness_response_period and fallback_owner data related to the calling safe
+    /// @dev Note: Disabling the module also cancels any ongoing challenges
+    function disable() external {
+        ModuleConfig storage config = safeConfigs[msg.sender];
+        // Check if the calling safe has the module enabled
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        // Erase the configuration data for this safe
+        delete safeConfigs[msg.sender];
+        // Also clear any active challenge
+        delete challengeStartTime[msg.sender];
+        emit ModuleDisabled(msg.sender);
+    }
+
+    /// @notice Returns challenge_start_time + liveness_response_period if there is a challenge for the given safe, or
+    /// 0 if not
+    /// @dev MUST never revert
+    /// @param _safe The Safe address to query
+    /// @return The challenge end timestamp, or 0 if no challenge
+    function isChallenged(address _safe) external view returns (uint256) {
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            return 0;
+        }
+        ModuleConfig storage config = safeConfigs[_safe];
+        return startTime + config.livenessResponsePeriod;
+    }
+
+    /// @notice Challenges an enabled safe
+    /// @dev MUST only be executable by fallback owner of the challenged safe
+    /// @dev MUST revert if there is a challenge for the safe
+    /// @dev MUST set challenge_start_time to the current block time
+    /// @dev MUST emit the ChallengeStarted event
+    /// @param _safe The Safe to challenge
+    function challenge(address _safe) external {
+        ModuleConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
```

Comment:
We also want to require that the module is enabled in the safe  with `safe.isModuleEnabled(address(this)))`

Reply:
Fixed in [f5fb1d3](https://github.com/ethereum-optimism/optimism/pull/17272/commits/f5fb1d39191c2494752f01f9539faf8bf1002e29)